scripts/west_commands/twister_cmd.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2022 Google
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import os
+import sys
+
+from west import log
+
+from zephyr_ext_common import Forceable
+from zephyr_ext_common import get_twister_path
+
+# Resolve path to twister libs and add imports
+twister_path = get_twister_path()
+os.environ["ZEPHYR_BASE"] = str(twister_path.parent)
+
+sys.path.insert(0, str(twister_path))
+sys.path.insert(0, str(twister_path / "pylib" / "twister"))
+
+from importlib.util import spec_from_loader, module_from_spec
+from importlib.machinery import SourceFileLoader
+
+spec = spec_from_loader(
+    "twister", SourceFileLoader("twister", str(twister_path / "twister"))
+)
+twister = module_from_spec(spec)
+spec.loader.exec_module(twister)
+
+from twisterlib.environment import add_parse_arguments, parse_arguments
```

Comment:
Hmm, giving this another look, why are you both using importlib **and** importing from twisterlib?

The importlib library is "heavy machinery" in my opinion and I think this may be a code smell.

Why can't the twister script be refactored a bit more so that the shared code you need can be imported from twisterlib using an ordinary import, instead of breaking out importlib?

Reply:
I tried `import twister` but since `scripts/twister` is missing the `*.py` extension, it errors when normally imported.  I can move the entire `main` definition to another file to avoid this?  I initially created `twister.py` soft link to `twister` but this failed compliance checks.  Searching the web lead me to this solution to import files that don't have the python extension.

Reply:
> I can move the entire main definition to another file to avoid this?

Right, that is what I am saying -- move the necessary shared code to a python file inside twisterlib, and import that from both `twister` and `twister_cmd.py`.

Reply:
(This would need approval from the twister maintainers, but it seems like the simplest way to handle this.)